src/routers/xcm/Router.sol

```diff
@@ -34,12 +34,31 @@ contract ConnectorXCMRouter is Router, Test {
         _;
     }
 
+    // TODO: re-add onlyCentrifugeChainOrigin
```

Comment:
I think we found out the authentication was not the problem and we can remove this `handleUnauthenticated` method again?